treeherder/log_parser/artifactbuilders.pyx

```diff
@@ -98,3 +108,60 @@ class BuildbotPerformanceDataArtifactBuilder(ArtifactBuilderBase):
             TalosParser()
         ]
         self.name = "talos_data"
+
+class StructuredLogArtifactBuilder(ArtifactBuilderBase):
+    """
+    Makes the stuctured log ``struct_faults`` artifact
+
+    """
+
+    def __init__(self, url=None):
+        """Construct artifact builder for the log viewer"""
+        super(StructuredLogArtifactBuilder, self).__init__(url)
+        self.name = "structured_log_summary"
+        self.url = url
+
+    class FaultSummaryHandler(object):
+        def __init__(self):
+            self.serial = 0
+            self.lines = []
+
+        def is_fault(self, message):
+            try:
+                retval = any(["level" in message and
+                    message["level"] in ("ERROR", "WARNING", "CRITICAL"),
+                    "expected" in message,
+                    message["action"] == "crash"])
+                return retval
+            except:
+                logger.warning("FaultSummaryHandler line exception {0}".format(message))
+                return False
+
+
+        def __call__(self, data):
+            self.serial += 1
+
+            if self.is_fault(data):
+                data['serial'] = self.serial
+                self.lines.append(data)
+
+            return None
+
+    def parse_log(self):
```

Comment:
This parsing works a little differently.  Because mozlog parses the whole file itself and the other parsers in ``parsers.pyx`` are line-by-line parsers.  We could rename them as ``LineParserXX`` or something like that.

I could dissect the way ``mozlog`` works to make this a more line-by-line parsing, but I didn't think it was a good idea to do that.